test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_fill.cpp

```diff
@@ -0,0 +1,461 @@
+                                       &pattern, sizeof(cl_int), 0, data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandFillBufferKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        error = clCommandFillImageKHR(command_buffer, nullptr, dst_image,
+                                      fill_color_1, origin, region, 0, nullptr,
+                                      nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandFillImageKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        context1 = clCreateContext(0, 1, &device, nullptr, nullptr, &error);
+        test_error(error, "Failed to create context");
+
+        out_mem_ctx =
+            clCreateBuffer(context1, CL_MEM_WRITE_ONLY,
+                           sizeof(cl_int) * num_elements, nullptr, &error);
+        test_error(error, "clCreateBuffer failed");
+
+        dst_image_ctx = create_image_2d(context1, CL_MEM_WRITE_ONLY, &formats,
+                                        512, 512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        queue1 = clCreateCommandQueue(context1, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+
+        dst_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    clContextWrapper context1;
+    clCommandQueueWrapper queue1;
+    clMemWrapper out_mem_ctx;
+    clMemWrapper dst_image;
+    clMemWrapper dst_image_ctx;
+    const cl_uint fill_color_1[4] = { 0x05, 0x05, 0x05, 0x05 };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+    const cl_int pattern = 0xFF;
+};
+
+// CL_INVALID_SYNC_POINT_WAIT_LIST_KHR if sync_point_wait_list is NULL and
+// num_sync_points_in_wait_list is > 0, or sync_point_wait_list is not NULL and
+// num_sync_points_in_wait_list is 0, or if synchronization-point objects in
+// sync_point_wait_list are not valid synchronization-points.
+struct CommandBufferCommandFillSyncPointsNullOrNumZero
+    : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_sync_point_khr invalid_point = 0;
+        std::vector<cl_sync_point_khr*> invalid_sync_points;
```

Comment:
Same comment as in another PR about not needing to create a vector here, and just passing `&invalid_point` directly to function.